woocommerce-myparcel.php

```diff
@@ -216,45 +229,162 @@ public function load_classes()
+
+            return [];
+        }
+
         /**
          * Check if woocommerce is activated
+         *
+         * @return bool
          */
-        public function is_woocommerce_activated()
+        public function isWooCommerceActivated(): bool
         {
-            $blog_plugins = get_option('active_plugins', []);
-            $site_plugins = get_site_option('active_sitewide_plugins', []);
+            return is_plugin_active('woocommerce/woocommerce.php');
+        }
 
-            if (in_array('woocommerce/woocommerce.php', $blog_plugins)
-                || isset($site_plugins['woocommerce/woocommerce.php'])) {
-                return true;
-            } else {
-                return false;
+        /**
+         * Notification for the minimum WooCommerce version
+         */
+        public function showRequiredWooCommerceNotification(): void
+        {
+            $message = $this->errorMessage['message'];
+            $button  = $this->errorMessage['button'];
+
+            if ($message) {
+                echo sprintf(
+                    '<div class="notice notice-error"><p><strong>%s</strong> %s</p><p>%s %s</p></div>',
+                    __('woocommerce_myparcel', 'woocommerce-myparcel'),
+                    $message,
+                    $button,
+                    $this->getDeactivateMyParcelButton()
+                );
             }
         }
 
         /**
-         * WooCommerce not active notice.
+         * @return array|null
          */
-        public function need_woocommerce()
+        public function checkInstalledWooCommerceVersion(): ?array
         {
-            $error = sprintf(
-                __("WooCommerce MyParcel requires %sWooCommerce%s to be installed & activated!",
-                    "woocommerce-myparcel"
-                ),
-                '<a href="http://wordpress.org/extend/plugins/woocommerce/">',
-                '</a>'
+            if (! empty($this->getWooCommerceData()) && ! $this->woocommerceVersionMeets()) {
+                $errorMessage = __("error_woocommerce_minimum_version", "woocommerce-myparcel");
+                $error        = str_replace('{woocommerce_version}', self::WOOCOMMERCE_VERSION_REQUIRED, $errorMessage);
+
+                $this->setAdminErrorMessage(__($error), null);
+            }
+
+            if (! $this->isWooCommerceActivated()) {
+                $this->setAdminErrorMessage(__('error_woocommerce_not_activated', 'woocommerce-myparcel'), $this->getActivateWooCommerceButton());
+            }
+
+            if (! $this->getWooCommerceData()) {
+                $this->setAdminErrorMessage(__('error_woocommerce_not_installed', 'woocommerce-myparcel'), $this->getInstallWooCommerceButton());
+            }
+
+            return $this->errorMessage;
+        }
+
+        /**
+         * @param string      $message
+         * @param string|null $button
+         */
+        public function setAdminErrorMessage(string $message, ?string $button): void
```

Comment:
Ik vind het toch wel (nog meer) een zootje worden, zou je al die admin notificatie logica in een nieuwe class kunnen trekken?

Reply:
Voor 3 meldingen een hele class maken vind ik zelf redelijk overkill

Reply:
Dat is wordpress mentaliteit 😛  Een bestand met 563 regels is onnodig complex

"3 meldingen" maar 13 methods die allemaal relateren aan compatibility checken, meldingen laten zien en acties daarop ondernemen:

```
deactivatePlugin
generateButtons
setAdminErrorMessage
phpVersionMeets
showRequiredPhpNotification
checkInstalledWooCommerceVersion
getActivateWooCommerceButton
getInstallWooCommerceButton
getWooCommerceData
isWooCommerceActivated
isWooCommerceInstalled
showRequiredWooCommerceNotification
woocommerceVersionMeets
```

Ik snap dat het wat meer werk is maar het is op lange termijn grote blij als we dingen eindelijk wat overzichtelijker en schoner gaan maken. Ik zie dit gewoon heel graag in een apart bestand omdat het allemaal chaos is die niet nodig is in `WCMYPA`.